project_generator/templates/makefile_armclang.tmpl

```diff
@@ -0,0 +1,33 @@
+{% extends "makefile.tmpl" %}
+
+{% block CC %}armclang{% endblock %}
+{% block CXX %}armclang{% endblock %}
+{% block AS %}armclang{% endblock %}
+{% block LD %}armlink{% endblock %}
+{% block AR %}armar{% endblock %}
+{% block CPP %}armclang{% endblock %}
+{% block OBJCOPY %}fromelf{% endblock %}
+{% block OBJDUMP %}fromelf{% endblock %}
+{% block SIZE %}fromelf{% endblock %}
+{% block NM %}fromelf{% endblock %}
+
+{% block TOBIN %}--bincombined{% endblock %}
+{% block TOHEX %}--i32combined{% endblock %}
+{% block objcopy_output %}--output{% endblock %}
+{% block objdump_output %}--output{% endblock %}
+{% block OBJDUMPFLAGS %}--text -acdyrz --interleave=source{% endblock %}
+{% block GENASMFLAGS %}-S -W{% endblock %}
+{% block SIZEFLAGS %}--text -z{% endblock %}
+{% block nm_output %}--output{% endblock %}
+{% block NMFLAGS %}--text -arz{% endblock %}
+
+{% block TARGET_EXE_EXT %}.axf{% endblock %}
+
+{% block COMMON_FLAGS %}--target=arm-arm-none-eabi -mcpu=$(CPU) -m$(INSTRUCTION_MODE){% endblock %}
+{% block CFLAGS %}$(COMMON_FLAGS) $(C_FLAGS) $(INC_DIRS_F) $(CC_SYMBOLS) -c -MMD{% endblock %}
+{% block CXXFLAGS %}$(COMMON_FLAGS) $(CXX_FLAGS) $(INC_DIRS_F) $(CC_SYMBOLS) -c -MMD{% endblock %}
+# Add -Wa,armasm,--show_cmdline to debug calls to armasm through armclang
+{% block ASFLAGS %}-masm=auto -x assembler-with-cpp $(COMMON_FLAGS) $(ASM_FLAGS) $(INC_DIRS_F) $(ASM_SYMBOLS) -c{% endblock %}
+{% block LD_OPTIONS %}--strict --scatter=$(LD_SCRIPT) --map --list $(OBJ_FOLDER)$(TARGET).map{% endblock %}
+{% block CPP_FLAGS %}-E -xc --target=arm-arm-none-eabi -mcpu=star -MMD{% endblock %}
```

Comment:
star ? Shouldn't this mcpu be set differently?

Reply:
Oh yeah... It was the last of the list and I read it as `*` but it seems is was a really bad choice ([https://developer.arm.com/tools-and-software/embedded/arm-compiler/downloads/version-6/6-13](see there)). We need to set a value otherwise we get this message `warning: 'armv4t' is unsupported in this version of the product
` ([see here](https://www.keil.com/support/man/docs/armclang_ref/armclang_ref_chr1392632801932.htm)). I will change that to `cortex-m4`.

Reply:
The "STAR" CPU is the v8-M core designed by Arm China specifically for the China market.

Reply:
I fixed that in [4cbfb16]

Reply:
why do we set it to cortex-m4 and not let a user to do that? we should use the same as on line 26: `-mcpu=$(CPU)`

Reply:
Done in [4935770]. Originally I wanted a setting that made it clear that this `-mcpu` setting was not relevant, but this way it is guaranteed to work.